vagrant/src/main/java/org/jclouds/vagrant/internal/VagrantNodeRegistry.java

```diff
@@ -73,36 +93,41 @@ public long getDelay(TimeUnit unit) {
          return unit.convert(expiryTime - timeSupplier.get(), TimeUnit.MILLISECONDS);
       }
    }
-   private Map<String, VagrantNode> nodes = new ConcurrentHashMap<String, VagrantNode>();
-   private DelayQueue<TerminatedNode> terminatedNodes = new DelayQueue<TerminatedNode>();
+
+   private final DelayQueue<TerminatedNode> terminatedNodes = new DelayQueue<TerminatedNode>();
+   private final Supplier<Map<String, VagrantNode>> nodes;
 
    private volatile long lastVacuumMs;
-   private Supplier<Long> timeSupplier;
+   private final Supplier<Long> timeSupplier;
 
    @Inject
-   VagrantNodeRegistry(@TimeStamp Supplier<Long> timeSupplier) {
+   VagrantNodeRegistry(@TimeStamp Supplier<Long> timeSupplier, Supplier<Collection<VagrantNode>> existingMachines) {
       this.timeSupplier = timeSupplier;
+      this.nodes = Suppliers.memoize(new ConcurrentWrapperSupplier(existingMachines));
```

Comment:
Would it make sense to configure this supplier (qualified with `@Memoized` if appropriate) in a Guice module? It looks more flexible and correct to get injected what you're gonna use instead of decorating the arguments?